packages/react/src/analytics/integrations/Forter/__tests__/Forter.test.ts

```diff
@@ -0,0 +1,209 @@
+  }),
+}));
+
+jest.mock(
+  '../forterTokenLoadedEventDetector',
+  () =>
+    class {
+      getForterToken = jest.fn(async () => await mockForterTokenLoadedPromise);
+    },
+);
+
+utils.logger.warn = jest.fn();
+
+describe('Forter', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return true on its shouldLoad method', () => {
+    expect(Forter.shouldLoad()).toBe(true);
+  });
+
+  it('should return an instance of Forter class in its createInstance method', () => {
+    expect(
+      Forter.createInstance(
+        { siteId: mockSiteId },
+        loadIntegrationData,
+        strippedDownAnalytics,
+      ),
+    ).toBeInstanceOf(Forter);
+  });
+
+  it('should return an error when siteId is not set on Forter Options.', () => {
+    // @ts-expect-error try loading forter without passing siteId
+    Forter.createInstance({}, loadIntegrationData, strippedDownAnalytics);
+
+    expect(utils.logger.warn).toHaveBeenCalledWith(
+      "[Forter] - `siteId` parameter was not provided in options. It's mandatory to load Forter Integration.",
+    );
+  });
+
+  it('should log a warn message if `origin` is not specified in options', () => {
+    Forter.createInstance(
+      { siteId: mockSiteId },
+      loadIntegrationData,
+      strippedDownAnalytics,
+    ) as Forter;
+
+    expect(utils.logger.warn).toHaveBeenCalledWith(
+      "[Forter] - `origin` parameter was not provided in options. It's advisable to provide an origin option to aid in debugging",
+    );
+  });
+
+  it('should post a message to Omnitracking with forter token when it is loaded', async () => {
+    const createEventMock = async function (
+      type: TrackTypesValues,
+      event: string,
+    ) {
+      return Promise.resolve({
+        ...eventData,
+        event,
+        type,
+      });
+    };
+
+    const forterInstance = Forter.createInstance(
+      {
+        siteId: mockSiteId,
+        origin: mockOrigin,
+      },
+      loadIntegrationData,
+      { ...strippedDownAnalytics, createEvent: createEventMock },
+    );
+    // @ts-expect-error accessing private property just for testing
+    const spy = jest.spyOn(forterInstance, 'resolveOnSetUserPromise');
+    // @ts-expect-error test data
+    const data: SetUserEventData = {};
+
+    forterInstance.onSetUser(data);
+
+    await mockPostTrackingsPromise;
+
+    expect(postTracking).toHaveBeenCalledWith(
+      expect.objectContaining({
+        tenantId: eventData.context.tenantId,
+        clientId: eventData.context.clientId,
+        correlationId: eventData.user.localId,
+        customerId: `${eventData.user.id}`,
+        event: 'PageAction',
+        parameters: expect.objectContaining({
+          clientTimestamp: expect.any(String),
+          tid: ForterTokenTid,
+          uuid: expect.any(String),
+          val: JSON.stringify({
+            forterTokenCookie: mockForterToken,
+            origin: mockOrigin,
+            // @ts-expect-error
```

Comment:
Same as https://github.com/Farfetch/blackout/pull/649/files#r1105523379, add a description for why you had to add this, please.